Disable legacy sequence parsing in RA.

pchote opened the pull request:
Like #7738, but for RA.  Only two more PRs after this one!

pchote committed:
Disable legacy sequence parsing in RA.

penev92 commented on mods/ra/sequences/misc.yaml:
```diff
@@ -206,7 +210,9 @@ moveflsh:
+		UseTilesetExtension: true
+		TilesetOverrides:
+			DESERT: TEMPERAT
 select:
```
New line above here?

reaperrr replied:
Agreed, that line removal looks wrong.

reaperrr commented on mods/ra/sequences/decorations.yaml:
```diff
@@ -1,4 +1,10 @@
+		TilesetOverrides:
+			DESERT: TEMPERAT
+			INTERIOR: TEMPERAT
+
```
This empty line needs to be removed.

reaperrr commented on mods/ra/sequences/decorations.yaml:
```diff
@@ -1,4 +1,10 @@
+			INTERIOR: TEMPERAT
+
 	idle:
 		Start: 0
```
All the ```Start: 0``` from here...

reaperrr commented on mods/ra/sequences/decorations.yaml:
```diff
@@ -449,211 +647,239 @@ brl3:
 boxes09:
-	idle:
+	idle: boxes09.int
 		Start: 0
```
...to here can be removed.

reaperrr commented on mods/ra/sequences/infantry.yaml:
```diff
@@ -60,11 +60,19 @@ e1:
 	die6: electro
 		Start: 0
```
I think ```Start: 0``` can be removed throughout this file.

reaperrr commented on mods/ra/sequences/misc.yaml:
```diff
@@ -76,6 +76,10 @@ explosion:
 		Start: 0
```
All the ```Start: 0``` in this file can go, too.

reaperrr commented on mods/ra/sequences/structures.yaml:
```diff
@@ -9,6 +9,9 @@ fcom:
 	bib: bib3
 		Start: 0
```
For consistency and cleanup, the ```Start: 0``` should go from this file, too, while you're at it.

reaperrr committed:
RA sequence cleanups.

Note: Removing Length: * from interior boxes and larvae as well as barrels
is intentional, boxes have just 1 frame, and frames 2+3 of larvae and
barrels are just duplicates of the first, added by Westwood to work around
damage stage issues.

Fixes fence palette on desert maps.

reaperrr committed:
Fix sniper electro death anim.

Closes #7742.

pchote commented:
Added fixes by @reaperrr.

obrakmann commented:
Looks ok to me. Played games on all tilesets except interior, tested the editor on all tilesets. :+1:

reaperrr commented:
Tested and looks good to me.

There are two minor hiccups remaining, I forgot to add that missing newline @penev92 pointed out and noticed that the tent no longer uses tileset-specific art.
But since there's currently no ```tent.tem``` the latter is not that quick to fix, therefore I'd rather deal with these in a follow-up myself so this pr's merge doesn't get delayed, in other words :+1:

Mailaender commented:
This also broke the palette for walls in the DESERT tileset in the editor:

![image](https://cloud.githubusercontent.com/assets/756669/6880478/71c7c328-d533-11e4-8ba4-6c15dfd07a02.png)

Mailaender commented:
Also broke resource placement on DESERT maps using the WinForms editor:

![image](https://cloud.githubusercontent.com/assets/756669/6880512/07a1bf1e-d536-11e4-8e6a-9851a772c4c9.png)

reaperrr commented:
I'll add the fixes to #7747.

reaperrr commented:
On second thought, the cleanest fix would probably be to make the editor ```TilesetOverrides```-aware.

pchote commented:
@reaperrr: the whole problem here is that the editor doesn't understand or use sequences at all. It uses a collection of hacks to find a .shp/.tem/etc with the same filename as the actor.  I will adjust #7749 to fix some of these issues.
